src/combinatorics/young_diagram.lean

```diff
@@ -123,4 +128,148 @@ end distrib_lattice
+  rw finset.mem_preimage at hj,
+  exact ⟨j, hj⟩,
+end
+
+/-- Length of a row of a Young diagram --/
+def row_len (μ : young_diagram) (i : ℕ) : ℕ := nat.find $ μ.exists_not_mem_row i
+
+lemma mem_iff_lt_row_len {μ : young_diagram} {i j : ℕ} : (i, j) ∈ μ ↔ j < μ.row_len i :=
+by { rw [row_len, nat.lt_find_iff], push_neg,
+     exact ⟨λ h _ hmj, μ.up_left_mem (by refl) hmj h, λ h, h _ (by refl)⟩ }
+
+lemma row_eq_prod {μ : young_diagram} {i : ℕ} : μ.row i = {i} ×ˢ (finset.range (μ.row_len i)) :=
+by { ext ⟨a, b⟩,
+     simp only [finset.mem_product, finset.mem_singleton, finset.mem_range,
+                mem_row_iff, mem_iff_lt_row_len, and_comm, and.congr_right_iff],
+     rintro rfl, refl }
+
+lemma row_len_eq_card (μ : young_diagram) {i : ℕ} : μ.row_len i = (μ.row i).card :=
+by simp [row_eq_prod]
+
+@[mono]
+lemma row_len_anti (μ : young_diagram) (i1 i2 : ℕ) (hi : i1 ≤ i2) : μ.row_len i2 ≤ μ.row_len i1 :=
+by { by_contra' h_lt, rw ← lt_self_iff_false (μ.row_len i1),
+     rw ← mem_iff_lt_row_len at h_lt ⊢,
+     exact μ.up_left_mem hi (by refl) h_lt }
+
+end rows
+
+section columns
+/-! ### Columns and column lengths of Young diagrams.
+
+This section has an identical API to the rows section. --/
+
+/-- The `j`-th column of a Young diagram consists of the cells whose second coordinate is `j`. --/
+def col (μ : young_diagram) (j : ℕ) : finset (ℕ × ℕ) := μ.cells.filter (λ c, c.snd = j)
+
+lemma mem_col_iff {μ : young_diagram} {j : ℕ} {c : ℕ × ℕ} : c ∈ μ.col j ↔ c ∈ μ ∧ c.snd = j :=
+by simp [col]
+
+protected lemma exists_not_mem_col (μ : young_diagram) (j : ℕ) : ∃ i, (i, j) ∉ μ.cells :=
+begin
+  obtain ⟨i, hi⟩ := infinite.exists_not_mem_finset
+    ((μ.cells).preimage (λ i, prod.mk i j) (λ _ _ _ _ h, by {cases h, refl})),
+  rw finset.mem_preimage at hi,
+  exact ⟨i, hi⟩,
+end
+
+/-- Length of a column of a Young diagram --/
+def col_len (μ : young_diagram) (j : ℕ) : ℕ := nat.find $ μ.exists_not_mem_col j
+
+lemma mem_iff_lt_col_len {μ : young_diagram} {i j : ℕ} : (i, j) ∈ μ ↔ i < μ.col_len j :=
+by { rw [col_len, nat.lt_find_iff], push_neg,
+     exact ⟨λ h _ hmj, μ.up_left_mem hmj (by refl) h, λ h, h _ (by refl)⟩ }
+
+lemma col_eq_prod {μ : young_diagram} {j : ℕ} : μ.col j = (finset.range (μ.col_len j)) ×ˢ {j} :=
+by { ext ⟨a, b⟩,
+     simp only [finset.mem_product, finset.mem_singleton, finset.mem_range,
+                mem_col_iff, mem_iff_lt_col_len, and_comm, and.congr_right_iff],
+     rintro rfl, refl }
+
+lemma col_len_eq_card (μ : young_diagram) {j : ℕ} : μ.col_len j = (μ.col j).card :=
+by simp [col_eq_prod]
+
+@[mono]
+lemma col_len_anti (μ : young_diagram) (j1 j2 : ℕ) (hj : j1 ≤ j2) : μ.col_len j2 ≤ μ.col_len j1 :=
+by { by_contra' h_lt, rw ← lt_self_iff_false (μ.col_len j1),
+     rw ← mem_iff_lt_col_len at h_lt ⊢,
+     exact μ.up_left_mem (by refl) hj h_lt }
+
+@[simp] lemma transpose_col_len (μ : young_diagram) (j : ℕ) : μ.transpose.col_len j = μ.row_len j :=
```

Comment:
There are some exceptions, but generally the naming convention follows the order for if you were to not use dot notation.
```suggestion
@[simp] lemma col_len_transpose (μ : young_diagram) (j : ℕ) : μ.transpose.col_len j = μ.row_len j :=
```